runtime/storage/sqldb/sqldb.go

```diff
@@ -262,6 +261,17 @@ func getPool(name string) *pgxpool.Pool {
 	if err != nil {
 		panic("sqldb: setup db: " + err.Error())
 	}
+	runtime.RegisterShutdown(func(force context.Context) {
+		done := make(chan struct{}, 1)
+		go func() {
+			pool.Close()
+			close(done)
+		}()
+		select {
```

Comment:
I get why here, but this is a bit of an annoying design to hve to do this for each registered shutdown function.

Reply:
Yeah it is. The alternative is for the `doShutdown` function to just ignore the fact that the shutdown handlers are still running and just proceed with shutting down after the timeout. That would be less code, but also worse? In this way the code can decide how to best shutdown gracefully.

Note that not every use case is doing this; the HTTP server doesn't need to. It depends on the semantics of the API. In the two sqldb-related cases there is no way of specifying the deadline and the methods block potentially forever, hence this code.